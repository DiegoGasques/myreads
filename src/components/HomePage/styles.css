.list-books {
  background-color: linen;
}
