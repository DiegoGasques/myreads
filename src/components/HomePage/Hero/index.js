import React from "react";

import "./styles.css";

function Hero() {
  return <div className="hero" />;
}

export default Hero;
