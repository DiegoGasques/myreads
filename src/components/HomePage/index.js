import React from "react";
import PropTypes from "prop-types";

import NavBar from "../NavBar";
import Logo from "../Logo";
import Container from "../Container";
import Hero from "./Hero";
import BookShelf from "../BookShelf";
import OpenSearchBtn from "./OpenSearchBtn";

import "./styles.css";

export function fromStatusKeyToTitle(str) {
  return str
    .split(/(?<!^)(?=[A-Z])/)
    .map(word => word[0].toUpperCase() + word.slice(1))
    .join(" ");
}

function HomePage({ books, keys }) {
  return (
    <div className="list-books">
      <NavBar>
        <Logo path="/" text="MyReads" size="md" />
      </NavBar>
      <Container>
        <Hero />
      </Container>

      <div className="list-books-content">
        <Container>
          {Object.keys(keys).map(k => {
            return (
              <BookShelf
                key={k}
                title={fromStatusKeyToTitle(keys[k])}
                filteredBooks={books.filter(b => b.shelf === keys[k])}
              />
            );
          })}
        </Container>
      </div>

      <OpenSearchBtn />
    </div>
  );
}

HomePage.propTypes = {
  books: PropTypes.arrayOf(PropTypes.object).isRequired,
  keys: PropTypes.object.isRequired
};

export default HomePage;
