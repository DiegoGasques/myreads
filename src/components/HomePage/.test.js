import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import HomePage, { fromStatusKeyToTitle } from "./index";
import BooksApp from "../App";

import { books } from "../../testFixtures";

it("Should match the component snapshot", () => {
  const wrapper = shallow(<HomePage books={books} keys={BooksApp.keys} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("Should test output given a shelf from book object", () => {
  const value = fromStatusKeyToTitle(books[0].shelf);
  expect(value).toBe("Currently Reading");
});
