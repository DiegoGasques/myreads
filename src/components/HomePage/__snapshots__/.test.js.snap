// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the component snapshot 1`] = `
<div
  className="list-books"
>
  <NavBar>
    <Logo
      path="/"
      size="md"
      text="MyReads"
    />
  </NavBar>
  <Container>
    <Hero />
  </Container>
  <div
    className="list-books-content"
  >
    <Container>
      <BookShelf
        filteredBooks={
          Array [
            Object {
              "authors": Array [
                "William E. Shotts, Jr.",
              ],
              "averageRating": 4,
              "description": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial 'shell shock,' you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust.",
              "id": "nggnmAEACAAJ",
              "imageLinks": Object {
                "smallThumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                "thumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
              },
              "pageCount": 480,
              "publishedDate": "2012",
              "publisher": "No Starch Press",
              "ratingsCount": 2,
              "shelf": "currentlyReading",
              "subtitle": "A Complete Introduction",
              "title": "The Linux Command Line",
            },
            Object {
              "authors": Array [
                "Liaquat Ahamed",
              ],
              "averageRating": 4.5,
              "description": "Argues that the stock market crash of 1929 and subsequent Depression occurred as a result of poor decisions on the part of four central bankers who jointly attempted to reconstruct international finance by reinstating the gold standard.",
              "id": "74XNzF_al3MC",
              "imageLinks": Object {
                "smallThumbnail": "http://books.google.com/books/content?id=74XNzF_al3MC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                "thumbnail": "http://books.google.com/books/content?id=74XNzF_al3MC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
              },
              "pageCount": 564,
              "publishedDate": "2009-01",
              "publisher": "Penguin",
              "ratingsCount": 14,
              "shelf": "currentlyReading",
              "subtitle": "The Bankers Who Broke the World",
              "title": "Lords of Finance",
            },
          ]
        }
        key="shelf1"
        title="Currently Reading"
      />
      <BookShelf
        filteredBooks={
          Array [
            Object {
              "authors": Array [
                "James Clifford",
              ],
              "averageRating": 5,
              "description": "Includes bibliographical references and index.",
              "id": "EDOVAT808fUC",
              "imageLinks": Object {
                "smallThumbnail": "http://books.google.com/books/content?id=EDOVAT808fUC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                "thumbnail": "http://books.google.com/books/content?id=EDOVAT808fUC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
              },
              "pageCount": 408,
              "publishedDate": "1997",
              "publisher": "Harvard University Press",
              "ratingsCount": 1,
              "shelf": "wantToRead",
              "subtitle": "Travel and Translation in the Late Twentieth Century",
              "title": "Routes",
            },
          ]
        }
        key="shelf2"
        title="Want To Read"
      />
      <BookShelf
        filteredBooks={Array []}
        key="shelf3"
        title="Read"
      />
    </Container>
  </div>
  <OpenSearchBtn />
</div>
`;
